### ISSUE TITLE
Anchor links in HTML5 mode go to 'base url' and trigger a navigation event

### ISSUE THREAD

**Issue by fwielstra** [angular/angular.js#5519]
I have an app with `$locationProvider.html5Mode(true)`, so no hashes in the URLs; I figured I could use anchors like they used to again, but unfortunately that doesn't seem to be the case. 

I made a plunkr to try and highlight the problem:

http://plnkr.co/edit/Upc89hCMMVbAhJaV2XYd?p=preview

full-screen with visible path version: http://run.plnkr.co/IA9Gwp52MzFpE1O6/

Issue 1 is that an `<a href="#anchor">` will link to `http://host/#anchor`, instead of `http://host/currenPath/#anchor`.

Issue 2 is that when I create an url `<a href="path/#anchor">` and click it, a navigation event is triggered - highlighted in the plunkr by a page transition animation.

StackOverflow questions offer answers [like this](http://stackoverflow.com/questions/14712223/how-to-handle-anchor-hash-linking-in-angularjs) work around the limitation by calling `$location.hash()` and `$anchorScroll`, but those will also trigger a navigation event. The second solution there is to reset $location.hash to the old value - i.e. none - to avoid the navigation event from triggering, but that removes the option to link directly to anchors from other pages and removes bookmarkability.

I came across issue #648 which seems to outline the same issue (two years ago), but it seems it stopped working sometime since then.

tl;dr: I'd like to do `<a href="#anchor">` that links to an anchor at the current route, but it links to the base URL and triggers a navigation event.

**Comment by eduncan26** [comment]
Any updates on this?

**Comment by jackhsu978** [comment]
I had the same issue. I found that it's because I had `<base href="/">`. Changing it to `<base href="">` fixes the problem. I tested on Chrome, Firefox, and Safari.

**Comment by divmgl** [comment]
That's interesting; `<base href="">` does not work for me at all. I'm getting some HTML5 provider errors.

**Comment by daniellavoie** [comment]
@vojtajina Thanks ! I had the same issue and you explanations made me double check my base href. Removing the front slash did it for me.

**Comment by morwin10** [comment]
I had a similar problem, I wanted to create links with Hash
Without that it will be removed by html5 mode, in the first time the page load
So :
$ LocationProvider.hashPrefix ( '#');
And it works !

That way you can create links with Hashes
even when html5 is enabled

**Comment by morwin10** [comment]
I had a similar problem, I wanted to create links with Hash
Without it will be removed by html5 mode, in the first time the page load
So :
$LocationProvider.hashPrefix ( '#');
And it works !

That way you can create links with Hashes
even when html5 is enabled

**Comment by bennlich** [comment]
@vojtajina The plunker doesn't seem to work for me. I added a log statement to the router controller so you can see that the controller is being re-instantiated every time you click. Is this expected?

http://plnkr.co/edit/a36L65c3vo8bFS0Bfszl?p=preview

**Comment by Narretz** [comment]
I think this boils down to a configuration issue. Also, I couldn't reproduce any flickering. See here http://plnkr.co/edit/rwct8raZ4xCpYfxImg9B?p=preview

**Comment by bennlich** [comment]
@Narretz Did you see my plnkr? Controller is getting reinstantiated every click.

http://plnkr.co/edit/a36L65c3vo8bFS0Bfszl?p=preview

**Comment by Narretz** [comment]
@bennlich strange, I thought my plunker didn't show the problem anymore. But it does now.
